lib/logger/lib/logger.ex

```diff
@@ -99,6 +99,22 @@ defmodule Logger do
   `:crash_reason`, `:initial_call`, and `:registered_name` are available
   only inside behaviours such as GenServer, Supervisor, and others.
 
+  For example, you might wish to include a custom error code in your logs:
+
+      Logger.error("We have a problem", [error_code: :pc_load_letter])
+
+  In your app's logger configuration, you would need to whitelist the
+  `:error_code` key and you would need to include `$metadata` as part of
+  your log format template:
+
+      config :logger, :console,
+       format: "$message[$level]$metadata\n",
+       metadata: [:error_code, :file]
+
+  Your logs might then receive lines like this:
+
+      We have a problem[error]error_code=pc_load_letter file=lib/app.ex
```

Comment:
While this is correct, it is a bit hard to read. What if the format is `[$level] $message $metadata`?

Reply:
Sounds good -- I just wanted to show that `$metadata` has to be in the formatting string.

Reply:
I think people is going to be more familiar with `"$metadata[$level] $message\n"` since that's what [ships by default with Phoenix](https://github.com/phoenixframework/phoenix/blob/20c80d9b3f661757ad847bf9abbf08bbbb8b8c16/installer/templates/phx_single/config/config.exs#L24).

In addition, I think it reads better because the log level kind of separates the message and the metadata:

```
error_code=pc_load_letter file=lib/app.ex [error] We have a problem 
```

Reply:
I pushed an update.  Again, the emphasis here is on how the metadata argument must line-up with the whitelisted metadata keys.  I don't think the level of confusion about the specifics of the formatting string (or the order of its arguments) is as high.